src/manifest.rs

```diff
@@ -52,23 +52,34 @@ struct ScaffoldData {
     settings: serde_yaml::Mapping,
 }
 
-pub(crate) fn manifest(uri: &str, resource_type: &str, settings: Option<String>) -> Result<()> {
-    let wasm_path = crate::utils::wasm_path(uri)?;
+pub(crate) fn manifest(id: &str, resource_type: &str, settings: Option<String>) -> Result<()> {
+    let wasm_path: PathBuf; 
+    let mut uri: String = id.to_string();
+
+    let is_sha = crate::utils::is_sha(id);
+    if is_sha {
+        uri = crate::utils::uri_from_sha(id)?;
+        wasm_path = crate::utils::wasm_path_from_sha(id)?;
+    } else {
+        let uri = crate::utils::map_path_to_uri(id)?;
+        wasm_path = crate::utils::wasm_path(uri.as_str())?;
+    }
```

Comment:
nitpicking, this code already exists inside of the inspect command. Maybe we could move it into a new dedicated function under `crate::utils`?